parser_library/src/workspace_manager.cpp

```diff
@@ -977,9 +963,10 @@ class workspace_manager_impl final : public workspace_manager,
 
     void add_workspace(const char* name, const char* uri) override
     {
+        auto normalized_uri = resource_location(uri).lexically_normal();
         auto& ows = m_workspaces
-                        .try_emplace(name,
-                            resource_location(std::move(uri)),
+                        .try_emplace(normalized_uri,
```

Comment:
Not sure about this change. Are we sure that it's working as advertised in the following scenarios?

- Usually, VS Code basically forbids me to open two identical workspaces (i.e. workspaces pointing to the same directory). However, sometimes I can manage to open them side by side.

- Multi-root Workspaces

Reply:
Nevertheless, I can see that the `name` variable is/was probably just a fancy name for a workspace URI?

Reply:
side-by-side - you mean 2 vscodes?

Reply:
Yes

Reply:
the biggest problem was that we were using URI to remove the workspace by name. so that definitely seems wrong. I could combine name & normalized_uri as the key, that should be fine an allow the same workspaces to be opened under with different names, however are workspace routine will be completely arbitrary, since we then choose a workspace by matching a file uri.

Reply:
ok, so two vscodes are not a problem - each should have its own language server instance

Reply:
If I write the following multi-root ws configuration file, the VS Code still shows only one root directory.
```json
{
	"folders": [
		{
			"path": "example_workspace"
		},
		{
			"path": "example_workspace"
		}
	],
	"settings": {
		"hlasm.serverVariant": "native"
	}
}
```

So I guess we're fine.

Reply:
Does it break if you refer to the same folder via a convoluted way "../parent/example_workspace"?

Reply:
Nope, still shows only a single directory